src/fstext/trivial-factor-weight.h

```diff
@@ -307,19 +292,18 @@ class TrivialFactorWeightFstImpl
 
   typedef unordered_map<Element, StateId, ElementKey, ElementEqual> ElementMap;
 
-  const Fst<A> *fst_;
+  std::unique_ptr<const Fst<A>> fst_;
   float delta_;
```

Comment:
Here make_shared initializes the underlying fst implementation not the unique pointer kept by that implementation.